### ISSUE TITLE
Build with older Qt5 fails

### ISSUE THREAD

**Issue by songo** [muse-sequencer/muse#567]
Hi,
I tried to build latest  MusE 3 version from git (3d8c0e482b4ddcd5d70e5d867a0426181c4d6c12) on Debian Jessie. It fails due to Qt compatibility issue:
```
/home/songo/src/debian/muse/muse-3.0.0git2017-05-22.1.3d8c0e4/muse/mixer/amixer.cpp: In member function ‘virtual 
QWidget* MusEGui::AudioMixerApp::setupComponentTabbing(QWidget*)’:
/home/songo/src/debian/muse/muse-3.0.0git2017-05-22.1.3d8c0e4/muse/mixer/amixer.cpp:859:31: error: ‘const struct 
QMetaObject’ has no member named ‘inherits’
     if(!widget->metaObject()->inherits(&Strip::staticMetaObject))
                               ^
muse/mixer/CMakeFiles/mixer.dir/build.make:96: recipe for target 'muse/mixer/CMakeFiles/mixer.dir/amixer.o' faile
d
make[3]: *** [muse/mixer/CMakeFiles/mixer.dir/amixer.o] Error 1
make[3]: Leaving directory '/home/songo/src/debian/muse/muse-3.0.0git2017-05-22.1.3d8c0e4/obj-x86_64-linux-gnu'
CMakeFiles/Makefile2:1303: recipe for target 'muse/mixer/CMakeFiles/mixer.dir/all' failed
make[2]: *** [muse/mixer/CMakeFiles/mixer.dir/all] Error 2
make[2]: Leaving directory '/home/songo/src/debian/muse/muse-3.0.0git2017-05-22.1.3d8c0e4/obj-x86_64-linux-gnu'
Makefile:140: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/songo/src/debian/muse/muse-3.0.0git2017-05-22.1.3d8c0e4/obj-x86_64-linux-gnu'
dh_auto_build: make -j1 returned exit code 2
debian/rules:6: recipe for target 'build' failed
make: *** [build] Error 2
```
dbcfb009551a1c8aafdb786e46502a5c74148bbf introduced this line:
```C++
if(!widget->metaObject()->inherits(&Strip::staticMetaObject))
      continue;
```
Unfortunately, `QMetaObject::inherits` is introduced in Qt 5.7 but only available Qt 5 version in Debian stable is 5.3.2. This is why build fails. Previous versions of MusE build on Debian just fine.

Any workaround?

**Comment by terminator356** [comment]
On 05/25/2017 06:58 AM, Krzysztof Gajdemski wrote:
>
> Hi,
> I tried to build latest MusE 3 version from git (3d8c0e4 
> <https://github.com/muse-sequencer/muse/commit/3d8c0e482b4ddcd5d70e5d867a0426181c4d6c12>) 
> on Debian Jessie. It fails due to Qt compatibility issue:
>
> |/home/songo/src/debian/muse/muse-3.0.0git2017-05-22.1.3d8c0e4/muse/mixer/amixer.cpp: 
> In member function ‘virtual QWidget* 
> MusEGui::AudioMixerApp::setupComponentTabbing(QWidget*)’: 
> /home/songo/src/debian/muse/muse-3.0.0git2017-05-22.1.3d8c0e4/muse/mixer/amixer.cpp:859:31: 
> error: ‘const struct QMetaObject’ has no member named ‘inherits’ 
> if(!widget->metaObject()->inherits(&Strip::staticMetaObject)) ^ 
> muse/mixer/CMakeFiles/mixer.dir/build.make:96: recipe for target 
> 'muse/mixer/CMakeFiles/mixer.dir/amixer.o' faile d make[3]: *** 
> [muse/mixer/CMakeFiles/mixer.dir/amixer.o] Error 1 make[3]: Leaving 
> directory 
> '/home/songo/src/debian/muse/muse-3.0.0git2017-05-22.1.3d8c0e4/obj-x86_64-linux-gnu' 
> CMakeFiles/Makefile2:1303: recipe for target 
> 'muse/mixer/CMakeFiles/mixer.dir/all' failed make[2]: *** 
> [muse/mixer/CMakeFiles/mixer.dir/all] Error 2 make[2]: Leaving 
> directory 
> '/home/songo/src/debian/muse/muse-3.0.0git2017-05-22.1.3d8c0e4/obj-x86_64-linux-gnu' 
> Makefile:140: recipe for target 'all' failed make[1]: *** [all] Error 
> 2 make[1]: Leaving directory 
> '/home/songo/src/debian/muse/muse-3.0.0git2017-05-22.1.3d8c0e4/obj-x86_64-linux-gnu' 
> dh_auto_build: make -j1 returned exit code 2 debian/rules:6: recipe 
> for target 'build' failed make: *** [build] Error 2 |
>
> dbcfb00 
> <https://github.com/muse-sequencer/muse/commit/dbcfb009551a1c8aafdb786e46502a5c74148bbf> 
> introduced this line:
>
> if(!widget->metaObject()->inherits(&Strip::staticMetaObject))
>        continue;
>
> Unfortunately, |QMetaObject::inherits| is introduced in Qt 5.7 but 
> only available Qt 5 version in Debian stable is 5.3.2. This is why 
> build fails. Previous versions of MusE build on Debian just fine.
>
> Any workaround?
>
>
Oops, sorry about that. Was introduced in last few commits.
I will try to find another solution. Stay tuned...
Tim.

**Comment by terminator356** [comment]
Oops, remove line. Next line qobject_cast() will take care of it.
Pushed to master. Thanks!
Tim.
